enterprise/backend/test/metabase_enterprise/serialization/api_test.clj

```diff
@@ -87,15 +89,19 @@
                          (tar-file-types f)))))
 
               (testing "On exception API returns log"
-                (with-redefs [u.compress/tgz (fn [& _] (throw (ex-info "[test] deliberate error message" {})))]
-                  (binding [api.serialization/*additive-logging* false]
-                    (let [res   (mt/user-http-request :crowberto :post 500 "ee/serialization/export" {}
-                                                      :collection (:id coll) :data_model false :settings false)
-                          lines (str/split-lines (slurp (io/input-stream res)))]
-                      (testing "First three lines for coll+dash+card, and then an error during compression"
-                        (is (= #{"Collection" "Dashboard" "Card"}
-                               (log-types (take 3 lines))))
-                        (is (re-find #"deliberate error message" (str/join "\n" (->> lines (drop 3) (take 3))))))))))
+                (let [extract-one serdes/extract-one]
+                  (with-redefs [serdes/extract-one (fn [model-name opts instance]
```

Comment:
I think we should stick to `mt/with-dynamic-redefs` unless there's a good reason not to. There's one gotcha however, you need to capture the value of the var here: `(mt/dynamic-value serdes/extract-one)`.